scripts/o3de/o3de/repo.py

```diff
@@ -18,34 +18,25 @@
 logger = logging.getLogger('o3de.repo')
 logging.basicConfig(format=utils.LOG_FORMAT)
 
+def get_repo_manifest_uri(repo_uri: str) -> str or None:
+    if not repo_uri:
+        logger.error(f'Repo URI cannot be empty.')
+        return None
 
-def process_add_o3de_repo(file_name: str or pathlib.Path,
-                          repo_set: set) -> int:
-    file_name = pathlib.Path(file_name).resolve()
-    if not validation.valid_o3de_repo_json(file_name):
-        logger.error(f'Repository JSON {file_name} could not be loaded or is missing required values')
-        return 1
-    cache_folder = manifest.get_o3de_cache_folder()
+    url = f'{repo_uri}/repo.json'
+    return url
 
-    repo_data = {}
-    with file_name.open('r') as f:
-        try:
-            repo_data = json.load(f)
-        except json.JSONDecodeError as e:
-            logger.error(f'{file_name} failed to load: {str(e)}')
-            return 1
+def download_repo_manifest(manifest_uri: str) -> pathlib.Path or None:
+    parsed_uri = urllib.parse.urlparse(manifest_uri)
+    repo_sha256 = hashlib.sha256(manifest_uri.encode())
+    cache_file = manifest.get_o3de_cache_folder() / str(repo_sha256.hexdigest() + '.json')
```

Comment:
nit: use f-strings instead of '+' operators on strings
```suggestion
    cache_file = manifest.get_o3de_cache_folder() / f'{repo_sha256.hexdigest()}.json')
```